### PR TITLE
Add JWT authentication & login page

### LINKED ISSUE #1: Authentication & Log In

### PR DESCRIPTION
**Pull request by Nolski**
# Auth with JWT

Resolves #1  - With this PR merged, we can manage user logins using JWT authentication.

As we do not have a user create page, you can make a new user by running the command on our API service `python manage.py createsuperuser`

Some fun screenshots below:

screenshot_2020-11-27-121120
screenshot_2020-11-27-121136

### CODE REVIEW

File: frontend/coalesce/src/pages/LogIn.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      </div>
+    </div>
+  </q-page>
+</template>
+
+<script>
+export default {
+  name: 'Login',
+  data () {
+    console.log('asdf')
```
Comment (stefanha) [inline frontend/coalesce/src/pages/LogIn.vue]: Minor nit: this looks like it was used temporarily for debugging and can now be removed

File: frontend/coalesce/src/store/auth/actions.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    const decoded = jwt_decode(token)
+    const exp = decoded.exp
+    const orig_iat = decoded.orig_iat
+
+    if (exp - (Date.now() / 1000) < 1800 && (Date.now() / 1000) - orig_iat < 628200) {
+      this.dispatch('refreshToken')
+    } else if (exp - (Date.now() / 1000) < 1800) {
+      // DO NOTHING, DO NOT REFRESH
+    } else {
+      // PROMPT USER TO RE-LOGIN, THIS ELSE CLAUSE COVERS THE CONDITION WHERE A TOKEN IS EXPIRED AS WELL
```
Comment (stefanha) [inline frontend/coalesce/src/store/auth/actions.js:47]: It's unclear to me whether this is a todo comment. Do we need to call a function to prompt the user to login again? (I'm not familiar with the framework, so maybe I'm missing something.)

**Comment by stefanha** [comment]
Looks good except for one question I posted.

File: frontend/coalesce/src/store/auth/actions.js (hunk 2 above)
Comment (Nolski) [reply frontend/coalesce/src/store/auth/actions.js:47]: That's correct, this is a TODO comment. I'll update this as I don't have our routing worked out yet.

**Pull request merged**